From 8d52998a04e6a1b65d95097e5ac27097b3a4923d Mon Sep 17 00:00:00 2001
From: Harish Krupo <harish.krupo.kps@intel.com>
Date: Mon, 11 Mar 2019 19:39:10 +0530
Subject: [PATCH 27/45] compositor.h: Add new renderer interfaces for HDR

Add set_output_{colorspace,hdr_metadata} interface for renderers so that
the compositor backend can inform the renderer of the expected target
color space and target hdr_metadata.

Signed-off-by: Harish Krupo <harish.krupo.kps@intel.com>
---
 include/libweston/libweston.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/libweston/libweston.h b/include/libweston/libweston.h
index 087c284b..e3167f26 100644
--- a/include/libweston/libweston.h
+++ b/include/libweston/libweston.h
@@ -927,6 +927,12 @@ struct weston_renderer {
 	void (*query_dmabuf_modifiers)(struct weston_compositor *ec,
 				int format, uint64_t **modifiers,
 				int *num_modifiers);
+
+	void (*set_output_colorspace)(struct weston_output *output,
+				      uint32_t colorspace);
+
+	void (*set_output_hdr_metadata)(struct weston_output *output,
+					struct weston_hdr_metadata *metadata);
 };
 
 enum weston_capability {
-- 
2.21.0

